gobblin-metastore/src/main/java/gobblin/metastore/StateStore.java

```diff
@@ -127,6 +127,18 @@ public T get(String storeName, String tableName, String stateId)
       throws IOException;
 
   /**
+   * Get a {@link State} with a given state ID from a the current table.
+   *
+   * @param storeName store name
+   * @param stateId state ID
+   * @return {@link State} with the given state ID or <em>null</em>
+   *         if the state with the given state ID does not exist
+   * @throws IOException
+   */
+  public T getCurrent(String storeName, String stateId)
```

Comment:
Are the two new `getCurrent...` methods really necessary? Plus, we are moving towards `DatasetStateStore`, and it has the access to the latest dataset state.